src/cachetools/__init__.py

```diff
@@ -526,20 +527,33 @@ def wrapper(*args, **kwargs):
 
         else:
 
+            _key_level_locks = {}
```

Comment:
Maybe better to use defaultdict to not allocate extra RLock?

Reply:
Interesting! That datatype looks ideal here. Out of curiosity, where is the extra RLock being allocated here?

Reply:
not here actually, but a pair lines below on `_key_level_lock = _key_level_locks.setdefault(k, RLock())`. As that line is under the lock so it must be atomic